zello-sdk/src/com/zello/sdk/Sdk.java

```diff
@@ -137,63 +137,68 @@ void onPause() {
 	//region Permissions
 
 	void requestVitalPermissions() {
-		Context context = _context;
+		requestVitalPermissions(null);
+	}
+
+	void requestVitalPermissions(@Nullable Activity activity) {
+		Context context = activity != null ? activity : _context;
 		String connectedPackage = _connectedPackage;
 		if (context == null || connectedPackage == null) {
 			return;
 		}
 		try {
 			Intent intent = new Intent();
-			intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+			if (activity == null) {
+				intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+			}
 			intent.setComponent(new ComponentName(connectedPackage, _pttPermissionsActivityClass));
 			intent.putExtra(Constants.EXTRA_REQUEST_VITAL_PERMISSIONS, true);
 			context.startActivity(intent);
 		} catch (Throwable ignored) {
 		}
 	}
 
-	void requestVitalPermissions(@Nullable Activity activity) {
-		String connectedPackage = _connectedPackage;
-		if (activity == null || connectedPackage == null) {
-			return;
-		}
-		try {
-			Intent intent = new Intent();
-			intent.setComponent(new ComponentName(connectedPackage, _pttPermissionsActivityClass));
-			intent.putExtra(Constants.EXTRA_REQUEST_VITAL_PERMISSIONS, true);
-			activity.startActivity(intent);
-		} catch (Throwable ignored) {
-		}
+	void requestLocationPermission() {
+		requestVitalPermissions(null);
 	}
 
-	void showMicrophonePermissionDialog() {
-		Context context = _context;
+	void requestLocationPermission(@Nullable Activity activity) {
```

Comment:
Maybe nitpicky, but this could be condensed even further by having a single private `requestPermissions` method that takes an `Activity` + a vararg String corresponding to the intent  boolean extras which are required. So then this method body would be `requestPermissions(activity, Constants.EXTRA_PERMISSION_LOCATION)`, and for microphone it would be `requestPermissions(activity, Constants.EXTRA_PERMISSION_DIALOG, Constants.EXTRA_PERMISSION_MICROPHONE)` etc.

All the public method signatures could remain the same, would only be eliminating some copy-pasted code,

Reply:
Thanks. Implemented.